src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+//! Query network interface addresses
+//!
+//! Uses the Linux and/or BSD specific function getifaddrs to query the list
+//! of interfaces and their associated addresses.
+
+use std::ffi;
+use std::iter::Iterator;
+use std::mem;
+use std::option::Option;
+
+use libc;
+use {Result, Error};
+
+use sys::socket::InetAddr;
+
+#[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
```

Comment:
nix doesn't support emscripten or fuschia, so you don't need to mention them here.  It doesn't hurt, though, so you can leave it in if you want to.  But why not enable it for the BSDs?  They support all this stuff.  If there are a few Linux specific flags, you can `#[cfg()]` individual flags instead of the whole thing.